apm-commons/apm-datacarrier/src/main/java/org/apache/skywalking/apm/commons/datacarrier/partition/SimpleRollingPartitioner.java

```diff
@@ -18,15 +18,17 @@
 
 package org.apache.skywalking.apm.commons.datacarrier.partition;
 
+import java.util.concurrent.atomic.AtomicInteger;
+
 /**
  * use normal int to rolling.
  */
 public class SimpleRollingPartitioner<T> implements IDataPartitioner<T> {
-    private volatile int i = 0;
+    private AtomicInteger i = new AtomicInteger(0);
```

Comment:
This will cause performance cost without much meaning. Rolling is for switching different channels, there is no need to be strict rule.